src/akushon/action/node/action_manager.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2021 Ichiro ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include <fstream>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "akushon/action/model/action_name.hpp"
+#include "akushon/action/node/action_manager.hpp"
+#include "akushon/action/process/interpolator.hpp"
+#include "nlohmann/json.hpp"
+#include "tachimawari/joint/joint.hpp"
+
+#include "stdlib.h"  // NOLINT
+#include "unistd.h"  // NOLINT
+
+namespace akushon
+{
+
+ActionManager::ActionManager()
+: actions({}), is_running(false)
+{
+  interpolator = std::make_shared<Interpolator>(std::vector<Action>(), Pose(""));
+}
+
+void ActionManager::insert_action(int action_id, const Action & action)
+{
+  actions.insert({action_id, action});
+}
+
+void ActionManager::delete_action(int action_id)
+{
+  actions.erase(action_id);
+}
+
+const Action & ActionManager::get_action(int action_id) const
+{
+  if (actions.find(action_id) != actions.end()) {
+    return actions.at(action_id);
+  }
+
+  return Action("");
+}
+
+void ActionManager::load_data(const std::string & path)
+{
+  for (auto [name, id] : ActionName::map) {
+    std::string file_name = path + "/action/" + name + ".json";
+    Action action = Action(name);
+
+    bool is_load_success = false;
+    try {
+      std::ifstream file(path);
+      nlohmann::json action_data = nlohmann::json::parse(file);
+
+      action.set_name(action_data["name"]);
+
+      for (auto [key, val] : action_data.items()) {
+        if (key.find("step_") != std::string::npos) {
```

Comment:
should be changed later, shouldn't strict the name of a pose with prefix `step_`